circuit_knitting_toolbox/circuit_cutting/cutting_decomposition.py

```diff
@@ -86,3 +93,123 @@ def partition_circuit_qubits(
         new_qc.data[i] = CircuitInstruction(qpd_gate, qubits=qubit_indices)
 
     return new_qc
+
+
+def decompose_gates(circuit: QuantumCircuit, gate_ids: Sequence[int]) -> QuantumCircuit:
+    r"""
+    Transform specified gates into :class:`TwoQubitQPDGate`\ s.
+
+    Args:
+        circuit: The circuit containing gates to be decomposed
+        gate_ids: The indices of the gates to decompose
+
+    Returns:
+        A copy of the input circuit with the specified gates replaced with :class:`TwoQubitGate`\ s
+    """
+    # Replace specified gates with TwoQubitQPDGates
+    new_qc = circuit.copy()
+    for gate_id in gate_ids:
+        gate = circuit.data[gate_id]
+        qubit_indices = [new_qc.find_bit(qubit).index for qubit in gate.qubits]
+        decomposition = QPDBasis.from_gate(gate.operation)
+        qpd_gate = TwoQubitQPDGate(decomposition, label=f"cut_{gate.operation.name}")
+        new_qc.data[gate_id] = CircuitInstruction(qpd_gate, qubits=qubit_indices)
+
+    return new_qc
+
+
+def partition_problem(
+    circuit: QuantumCircuit,
+    partition_labels: Sequence[str | int],
```

Comment:
I believe we decided at some point that `str | int` should be replaced with `Hashable`.  (I will make this comment only here, but there are 11 instances throughout this PR.)

Reply:
Not all hashables are sortable, and in order to be consistent with ordering from our subcircuit/subobservable dicts,  we sort by partition name often.

I think it's fine for the underlying package to be more general and for us to restrict the inputs through our interface for this reason, but open to suggestions

Reply:
Sorting will also break if they pass in a mix of strings and ints

Reply:
I am still hoping that it's possible to remove the sort and broaden this type.  I will follow up more here when I better understand the implications of this.

Reply:
I'm going to resolve this because I think it is out of scope for this release.

That being said, what we have now is *more* strict than you think is necessary, so loosening this to all `Hashable` in a future release shouldn't break anything.